message/validation/genesis/message_counts.go

```diff
@@ -56,13 +56,14 @@ func (c *MessageCounts) ValidateConsensusMessage(msg *specqbft.SignedMessage, li
 				err.got = fmt.Sprintf("commit, having %v", c.String())
 				return err
 			}
-		}
-		if len(msg.Signers) > 1 {
+		} else if len(msg.Signers) > 1 {
 			if c.Decided >= limits.Decided {
 				err := ErrTooManySameTypeMessagesPerRound
 				err.got = fmt.Sprintf("decided, having %v", c.String())
 				return err
 			}
+		} else {
+			return ErrNoSigners
```

Comment:
If we do multiple checks of the same things, perhaps it will be complex to follow. We already check this in `validConsensusSigners`

Reply:
I guess having the assumption is fine. But I also think it's good to make these assertions, even when they should be impossible due to previous logic. What do you guys think @y0sher @moshe-blox ?

Reply:
If we decide to make them, I think we'll need to make them consistent and add to all methods

Reply:
I agree with @nkryuchkov here. either we set some basic assertions we do all the time, or we count on previous code to have done this assertions.
I think we should leave it only in the most proper place and go forward assuming it was already checked. (as long as we know all messages go through this flow. )

Reply:
I think we can return an error but we need to emphasize this is a not expected "bug" error and not a common error that can happen with a valid message. maybe even panic or fatal here. 
we should comment about this case and use specialized error.

Reply:
I agree, but I'd prefer fatal

Reply:
Since I'm taking over this PR, I'm going to delete this because in my opinion if we add assertion checks, this is not the only place to do it, so we could do it in a new PR